pinc/ProjectState.inc

```diff
@@ -28,39 +28,48 @@ class ProjectState
 
 class ProjectStates
 {
-    private static $states = [];
+    /** @var array<string, ProjectState> */
+    private static array $states = [];
 
-    public static function add_state($state)
+    public static function add_state(ProjectState $state): void
     {
         self::$states[$state->name] = $state;
     }
 
-    public static function get_states()
+    /** @return string[] */
+    public static function get_states(): array
     {
         return array_keys(self::$states);
     }
 
-    public static function get_all()
+    /** @return array<string, ProjectState> */
+    public static function get_all(): array
     {
         return self::$states;
     }
 
-    public static function get_medium_label($state)
+    // TODO: We should ensure that `$state` is never null. For now, we allow it.
```

Comment:
I think this is a good goal but I'm not sure we'll get there and thus that the comment is useful. We've played fast-and-loose with data types -- particularly null / empty string -- for 20 years and if we started adding comments like this around it would be in 75% of all of the cases where we pass strings into functions.

With that sad and somewhat fatalistic view I don't think we should include these comments.

Reply:
Yeah, fair point about those comments.

I have seen cases of types being improved over time but it's a lot of work and we're nowhere close to having enough type coverage, at which point such changes are easier to do...

Removed those comments and kept the one about the return type as it seemed more of a one-off. Happy to remove it too and stick to documenting our current types rather than thinking too much ahead.

Reply:
The work you are doing is a great foundation for us making further type improvements. I'm hoping after we have more type coverage (and more code coverage via smoke tests?) we can make broader changes like `?string` to `string` in place like this.

One-offs like return comment types are great and welcome!